tests/50federation/40publicroomlist.pl

```diff
@@ -63,38 +63,32 @@
                   my $topic = $room->{topic};
                   my $canonical_alias = $room->{canonical_alias};
 
-                  my $aliases = $room->{aliases};
-                  defined $aliases or next;
+                  foreach my $alias_local (keys %rooms) {
+                     $canonical_alias =~ m/^\Q#$alias_local:\E/ or next;
 
-                  foreach my $alias (@{$aliases}) {
-                     foreach my $alias_local (keys %rooms) {
-                        $alias =~ m/^\Q#$alias_local:\E/ or next;
+                     my $room_config = $rooms{$alias_local};
 
-                        my $room_config = $rooms{$alias_local};
+                     log_if_fail "Alias", $alias_local;
+                     log_if_fail "Room", $room;
 
-                        log_if_fail "Alias", $alias_local;
-                        log_if_fail "Room", $room;
+                     assert_eq($room->{num_joined_members}, 1, "num_joined_members");
 
-                        assert_eq($canonical_alias, $alias, "canonical alias");
```

Comment:
I think it is worth pointing on that this assertion was removed entirely, I do not think there's anything to replace it with.